routing/base/followed_polyline.cpp

```diff
@@ -128,32 +133,36 @@ Iter FollowedPolyline::GetBestProjection(m2::RectD const & posRect,
   return GetClosestProjectionInInterval(posRect, distFn, hoppingBorderIdx, m_nextCheckpointIndex);
 }
 
-FollowedPolyline::UpdatedProjection FollowedPolyline::GetBestMatchingProjection(m2::RectD const & posRect) const
+Iter FollowedPolyline::GetBestMatchingProjection(m2::RectD const & posRect) const
 {
   CHECK_EQUAL(m_segProj.size() + 1, m_poly.GetSize(), ());
   // At first trying to find a projection to two closest route segments of route which is close
   // enough to |posRect| center. If |m_current| is right before intermediate point we can get |closestIter|
   // right after intermediate point (in next subroute).
   size_t const hoppingBorderIdx = min(m_segProj.size(), m_current.m_ind + 3);
-  auto const res = GetClosestMatchingProjectionInInterval(posRect, m_current.m_ind, hoppingBorderIdx);
-  if (res.m_iter.IsValid() || res.m_closerToUnmatching)
-    return UpdatedProjection{res.m_iter, res.m_closerToUnmatching};
+  auto const iter = GetClosestMatchingProjectionInInterval(posRect, m_current.m_ind, hoppingBorderIdx);
+  if (iter.IsValid())
+    return iter;
```

Comment:
`|closestIter|` in comment, but auto const `iter = GetClosestMatchingProjectionInInterval` in the code.

Reply:
fixed